pkg/util/iptables/iptables.go

```diff
@@ -784,3 +786,74 @@ func isResourceError(err error) bool {
 	}
 	return false
 }
+
+type ParseError interface {
+	Line() int
+	Error() string
+}
+
+type parseError struct {
+	cmd  string
+	line int
+}
+
+func (e parseError) Line() int {
+	return e.line
+}
+
+func (e parseError) Error() string {
+	return fmt.Sprintf("%s: input error on line %d: ", e.cmd, e.line)
+}
+
+// ParseRestoreError extracts the line from the error, if it matches returns parseError
+// for example:
+// input err: iptables-restore: line 51 failed
+// output parseError:  cmd = iptables-restore, line = 51
+func ParseRestoreError(err error) (ParseError, bool) {
+	errInfos := strings.Split(err.Error(), ":")
+	if len(errInfos) != 2 {
+		return nil, false
+	}
+
+	lineInfo := strings.TrimSpace(errInfos[1])
+	lineInfos := strings.Split(lineInfo, " ")
+	if len(lineInfos) != 3 {
+		return nil, false
+	}
+
+	line, errMsg := strconv.Atoi(lineInfos[1])
+	if errMsg != nil {
+		return nil, false
+	}
+	return parseError{cmd: lineInfos[1], line: line}, true
+}
+
+const delimiter = 0x0a
```

Comment:
ascii new line?
```
(nl)   10 0012 0x0a   New Line
```
if so we should add a comment explaining it

Reply:
maybe we should all use acsii to represent these separators(`":"、" "、"\n"`) and add a comment explaining it?